### PR TITLE
Bugfix FXIOS-8790 Bugzilla 1874560

### PR DESCRIPTION
**Pull request by razvanlitianu**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-8790)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/TODO)

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: focus-ios/Blockzilla/Modules/WebView/LegacyWebViewController.swift [hunk 1]
```diff
@@ -535,15 +535,15 @@ extension LegacyWebViewController: WKUIDelegate {
...
         guard navigationAction.targetFrame == nil else { return nil }
 
-        // validate the URL using URIFixup
-        guard let urlString = navigationAction.request.url?.absoluteString,
-              URIFixup.getURL(entry: urlString) != nil else {
-            // URL failed validation, prevent loading
+        // If URL is a file:// when web application calls window.open() or fails validation, prevent loading
+        guard let url = navigationAction.request.url,
+              let validatedURL = URIFixup.getURL(entry: url.absoluteString),
+              url.scheme != "file" else {
```
Comment (mattreaganmozilla) [inline focus-ios/Blockzilla/Modules/WebView/LegacyWebViewController.swift:541]: Do we know if there are any scenarios in which the `file://` scheme should be allowed? I'm not familiar with all of the potential use cases for that.

**Review by mattreaganmozilla — APPROVED** [review]
Would be good to get additional 👀 on this from @nbhasin2 but the change LGTM.

File: focus-ios/Blockzilla/Modules/WebView/LegacyWebViewController.swift (hunk 1 above)
Comment (nbhasin2) [reply focus-ios/Blockzilla/Modules/WebView/LegacyWebViewController.swift:541]: Hi, wouldn't this stop from opening file type? 

This solution basically stops the file from opening a new tab but that isn't something we should prevent. Instead we should be looking to make sure that the url bar has the correct address from the file `open` method.

**Review by nbhasin2 — CHANGES_REQUESTED** [review]
@razvanlitianu good approach although I don't feel we should block the users from opening new tab or same tab. Would it be possible to make sure the url has the correct address instead?

Requesting changes in the mean time related to above comment

File: focus-ios/Blockzilla/Modules/WebView/LegacyWebViewController.swift (hunk 1 above)
Comment (razvanlitianu) [reply focus-ios/Blockzilla/Modules/WebView/LegacyWebViewController.swift:541]: hey @nbhasin2 I'm also not familiar how file:// urls should behave on Focus. Do we have a use case for Firefox? I guess my question is how we can differentiate between a malicious file url and a good one.

**Comment by razvanlitianu** [comment]
Following a discussion with @nbhasin2, we decided to implement a solution similar to one used in JavaScript, which involves handling file URLs by searching through the default browser engine.

File: focus-ios/Blockzilla/Utilities/URIFixup.swift [hunk 2]
```diff
@@ -15,6 +15,14 @@ class URIFixup {
             return nil
         }
 
+        // Check if the URL string does not start with "file://". 
+        // If it starts with "file://", return nil, indicating that this function
+        // does not handle local file URLs, resulting in a google search
```
Comment (nbhasin2) [inline focus-ios/Blockzilla/Utilities/URIFixup.swift:20]:
```suggestion
        // does not handle local file URLs, resulting in a search
```

Removed google as a user might have another search engine selected

**Review by nbhasin2 — APPROVED** [review]

**Comment by razvanlitianu** [comment]
@Mergifyio backport release/v126

**Pull request closed without merging**